### DISCUSSION TITLE
[Q&A] 使用Azure openai key后，GPT-4无法使用插件功能

### DISCUSSION THREAD

**Discussion by liliwen365** [lobehub/lobehub#970] [General Question] [answered]
### 💻 系统环境 | Operating System

Windows

### 🌐 浏览器 | Browser

Chrome

### 🐛 问题描述 | Bug Description

使用Azure openai key后，GPT-4无法使用插件功能，GPT3.5可以使用插件功能；GPT-4使用插件后，正常对话也会报错，不使用插件，正常能够文本对话；报错信息如下：
![image](https://github.com/lobehub/lobe-chat/assets/76478019/5a6588d1-2844-43de-8cf9-86367fa4d0b7)
{
  "endpoint": "https://llwswedencentral.openai.azure.com/openai/deployments/gpt-4",
  "error": {
    "message": "Unrecognized request argument supplied: tools",
    "type": "invalid_request_error",
    "param": null,
    "code": null
  }
}

### 🚦 期望结果 | Expected Behavior

用Azure openai 的key使用gpt-4 不能使用插件，导致不能联网，希望解决！

[a comment by an automated account is not shown]

**Comment by showifconfig** [comment]
请教一下，您的配置，我Azure openai 3.5也无法使用，报错与你的4.0一致

[a comment by an automated account is not shown]

**Comment by arvinxx** [comment]
azure 的 openai 不支持 [tool_calls](https://github.com/lobehub/lobe-chat/issues/592) 吗…

**Comment by liliwen365** [comment]
> 请教一下，您的配置，我Azure openai 3.5也无法使用，报错与您的4.0一致
按Azure openai key 的配置，在初始配置时输入一个openaI的testkey，任意输入，部署完成后；在vercel中项目设置中添加环境变量environment-variables，将下图环境变量配置进去后，再重新部署
![image](https://github.com/lobehub/lobe-chat/assets/76478019/660de268-3a85-49de-9ca7-f992a42f067c)
重新部署页面如下：
![image](https://github.com/lobehub/lobe-chat/assets/76478019/db15db89-f1eb-49c8-a592-0ebe948c8788)

[a comment by an automated account is not shown]

**Comment by showifconfig** [comment]
感谢您的回答，可能我是使用One-api配合这个使用，所以导致的无法调用，我如果不开启插件的话，功能正常使用的

[a comment by an automated account is not shown]

**Comment by WeihanLi** [comment]
遇到同样的问题， 3.5 也不行

```json
{
  "endpoint": "",
  "error": {
    "message": "Unrecognized request argument supplied: tools",
    "type": "invalid_request_error",
    "param": null,
    "code": null
  }
}
```

**Comment by arvinxx** [comment]
@WeihanLi 你部署的可能是老版本的 gpt-3.5，试试使用最新的 version (2023-12) 部署一下

[a comment by an automated account is not shown]

**Comment by WeihanLi** [comment]
@arvinxx 感谢回复，怎么用最新的版本（2023-12）呀，好像没看到，能看到的最新的版本是 0613，有区域的限制吗？

![image](https://github.com/lobehub/lobe-chat/assets/7604648/9caec808-42e2-42c5-9ea9-c7ab51dc3892)

**Comment by WeihanLi** [comment, marked as answer]
@arvinxx 更新成 0613 之后可以了，感谢感谢

[a comment by an automated account is not shown]
